pkg/scm/git/git.go

```diff
@@ -62,18 +67,136 @@ func stringInSlice(s string, slice []string) bool {
 // ValidCloneSpec determines if the given string reference points to a valid git
 // repository
 func (h *stiGit) ValidCloneSpec(source string) bool {
-	url, err := url.Parse(source)
-	if err != nil {
+	rc, _, _, _, _, _ := validCloneSpec(source, false, nil)
+	return rc
+}
+
+// ValidCloneSpecRemoteOnly determines if the given string reference points to a valid remote git
+// repository; valid local repository specs will result in a return of false
+func (h *stiGit) ValidCloneSpecRemoteOnly(source string) bool {
+	rc, _, _, _, _, _ := validCloneSpec(source, true, nil)
+	return rc
+}
+
+// MungeNoProtocolURL will take a URL returned from net/url.Parse and make
+// corrections to the URL when no protocol is specified in the Scheme
+func (h *stiGit) MungeNoProtocolURL(source string, uri *url.URL) bool {
+	if uri == nil {
+		return false
+	}
+	rc, scheme, user, host, path, ref := validCloneSpec(source, false, uri)
+	if !rc {
+		return false
+	}
+	if scheme != "ssh" && scheme != "file" {
```

Comment:
yeah, the boolean return was in fact a placeholder for future use - its actual use in my corresponding origin change does not leverage it... certainly with lack of use I did not think it through.

I'm inclined to switch gears and have a void return.  Will do that as part of next push.

On comments, yeah, I had some on the origin side but not here.  I'll address.

On the file and ssh pairing, just to be clear, there "association" such as it is stems from the fact that from a git clone spec perspective, there are valid url's where neither file:// or ssh:// are specified, but based on other characters being present (or not present) the file:// or ssh:// can be derived.